lang/src/org/partiql/lang/planner/transforms/optimizations/RemoveUselessFilters.kt

```diff
@@ -0,0 +1,26 @@
+package org.partiql.lang.planner.transforms.optimizations
+
+import org.partiql.lang.domains.PartiqlPhysical
+import org.partiql.lang.errors.ProblemHandler
+import org.partiql.lang.planner.PartiqlPhysicalPass
+import org.partiql.lang.planner.transforms.isLitTrue
+
+/** Creates a pass that removes any `(filter ...)` where the predicate is simply `(lit true)`. */
```

Comment:
Really, removing useless `and`s is a limited form of constant folding.  I would expect `or`s and all other expressions with operands whose value is known at compile-time to be handled by a more generic constant folding pass which is out of scope for the time being.

A filter who's predicate is simply `false` is not truly "useless"--it always returns no rows!  It could be rewritten into some operator that returns an empty relation, but that's not within my scope currently.